system/httplib/httplib.h

```diff
@@ -194,6 +194,10 @@ using socket_t = int;
 #include <thread>
 
 #ifdef CPPHTTPLIB_OPENSSL_SUPPORT
+
+#if defined(_USE_OPENSSLV3)
+    #define OPENSSL_API_COMPAT 0x10100000L
+#endif
```

Comment:
It doesn't look right, to redefine this (in these PR changes) 9 times, including in 2  @headers.
I'd expect it to be defined once and be included from a common header.
Also not sure where/what uses it, why it's defined (needs comment).